org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/CompletionHandlerTest.java

```diff
@@ -1178,6 +1180,99 @@ public void testCompletion_testClassesAvailableIntoTestCode() throws Exception {
 		assertEquals("AbstractTest - foo.bar", list.getItems().get(0).getLabel());
 	}
 
+	@Test
+	public void testCompletion_overwrite() throws Exception {
+		ICompilationUnit unit = getCompletionOverwriteReplaceUnit();
+		//@formatter:on
+		int[] loc = findCompletionLocation(unit, "method(t.");
+		CompletionList list = server.completion(JsonMessageHelper.getParams(createCompletionRequest(unit, loc[0], loc[1]))).join().getRight();
+		assertNotNull(list);
+		CompletionItem ci = list.getItems().stream().filter(item -> item.getLabel().startsWith("testInt : int")).findFirst().orElse(null);
+		assertNotNull(ci);
+		assertEquals("testInt", ci.getInsertText());
+		assertEquals(CompletionItemKind.Field, ci.getKind());
+		assertEquals("999998554", ci.getSortText());
+		assertNull(ci.getTextEdit());
+		CompletionItem resolvedItem = server.resolveCompletionItem(ci).join();
+		assertNotNull(resolvedItem.getTextEdit());
+		List<TextEdit> edits = new ArrayList<>();
+		edits.add(resolvedItem.getTextEdit());
+		String returned = TextEditUtil.apply(unit, edits);
+		//@formatter:off
+			String expected =
+				"package foo.bar;\n\n" +
+				"public class BaseTest {\n" +
+				"    public int testInt;\n\n" +
+				"    public boolean method(int x, int y, int z) {\n" +
+				"        return true;\n" +
+				"    } \n\n" +
+				"    public void update() {\n" +
+				"        BaseTest t = new BaseTest();\n" +
+				"        t.method(t.testIntthis.testInt, this.testInt);\n" +
```

Comment:
you're testing the "overwrite" case, so the expectation should be `t.method(t.testInt, this.testInt)`